### ISSUE TITLE
MVVM Cross Support

### ISSUE THREAD

**Issue by jeromelaban** [unoplatform/uno#1250]
## What would you like to be added:

MVVM Cross support.

## For which Platform:

- [x] iOS
- [x] Android
- [x] WebAssembly
- [x] Windows

## Anything else we need to know?
Ongoing work is available here: https://github.com/nickrandolph/MvvmCross/tree/feature/unosupport2

**Comment by davidjohnoliver** [comment]
For now, using MvvmCross with Uno on WebAssembly requires setting the following somewhere in app code (eg from `App.xaml.cs` constructor):

```csharp
#if __WASM__
            Windows.UI.Core.CoreDispatcher.HasThreadAccessOverride = true;
#endif
```

Otherwise MvvmCross' `ExecuteOnMainThreadAsync()` method will deadlock (because Uno's `CoreDispatcher.HasThreadAccess` will always return `false`). 

This is a temporary requirement, until WASM threading is supported on all browsers and by Mono (https://github.com/mono/mono/issues/12453).

**Comment by nickrandolph** [comment]
@davidjohnoliver I've just updated the Uno bits in mvvmcross to Uno v3. I assume that the threading workaround is still required? Or should MvvmCross work without the HasThreadAccessOverride being set to true?

**Comment by davidjohnoliver** [comment]
Yes the workaround is still required. There's been some progress on threading support on Mono, but it's still at an experimental state and it's not yet enabled in Uno.UI.Wasm.

**Comment by MartinZikmund** [comment]
May be worth keeping open

**Comment by MartinZikmund** [comment]
# Repro generator

This issue was classified as not reproducible with a code sample or test.

**Reason:** This is a third-party library integration request for MvvmCross support. The work primarily involves changes to MvvmCross itself (as indicated by the linked MvvmCross PR), not Uno Platform's core. There is no specific Uno behavior to reproduce with a runtime test or sample.
